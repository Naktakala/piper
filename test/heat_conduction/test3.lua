nodes = {}
N = 100
xmin = 0.0
L = 1.0
nodes = {}

dx = L / N
for i = 0, N do
  nodes[i + 1] = xmin + dx * i
end
meshgen1 = chi_mesh.OrthogonalMeshGenerator.Create({ node_sets = { nodes, nodes } })
chi_mesh.MeshGenerator.Execute(meshgen1)

hcsystem = hcm.HeatConductionSystem.Create({
  kernels = {
    { type = hcm.ThermalConductionKernel.type, k = 16.0 },
    { type = chi_math.SinkSourceFEMKernel.type, value = 100.0e2 }
  },
  bcs = {
    {
      type = chi_math.FEMDirichletBC.type,
      boundaries = { "XMIN", "YMIN", "YMAX" }
    },
    {
      type = hcm.ConvectiveHeatFluxBC.type,
      boundaries = { "XMAX" },
      T_bulk = 100.0,
      convection_coefficient = 10000.0
    }
  }
})

phys1 = hcm.HCSteadyExecutor.Create({
  conduction_system = hcsystem,
  solver_params =
  {
    nl_method = "PJFNK",
    --nl_method = "NEWTON",
    l_rel_tol = 1.0e-5,
    --pc_options =
    --{
    --  pc_type = "hypre",
    --  pc_hypre_type = "boomeramg",
    --  pc_hypre_boomeramg_coarsen_type = "HMIS"
    --}
  }
})

chiSolverInitialize(phys1)
chiSolverExecute(phys1)

--############################################### PostProcessors
chi.CellVolumeIntegralPostProcessor.Create
({
  name = "avgval",
  field_function = "T",
  compute_volume_average = true
})
chi.ExecutePostProcessors({"avgval"})

if (master_export == nil) then
  chiExportMultiFieldFunctionToVTK({ "T" }, "test3")
end