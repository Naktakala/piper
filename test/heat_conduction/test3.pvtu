<VTKFile type="PUnstructuredGrid" version="0.1" byte_order="LittleEndian" header_type="UInt32" compressor="vtkZLibDataCompressor">
  <PUnstructuredGrid GhostLevel="0">
    <PPointData>
      <PDataArray type="Float64" Name="T"/>
    </PPointData>
    <PCellData>
      <PDataArray type="Int32" Name="Material"/>
      <PDataArray type="UInt32" Name="Partition"/>
      <PDataArray type="Float64" Name="T"/>
    </PCellData>
    <PPoints>
      <PDataArray type="Float64" Name="Points" NumberOfComponents="3"/>
    </PPoints>
    <Piece Source="test3_0.vtu"/>
    <Piece Source="test3_1.vtu"/>
  </PUnstructuredGrid>
</VTKFile>
