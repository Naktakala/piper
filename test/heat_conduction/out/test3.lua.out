mpiexec -np 2 /Users/vermjc/Repos/ChiTech/piper/bin/piper test3.lua --suppress_color --supress_beg_end_timelog master_export=false 
[0]  Parsing argument 1 test3.lua
[0m[0]  Parsing argument 2 --suppress_color
[0m[0]  Parsing argument 3 --supress_beg_end_timelog
[0]  Parsing argument 4 master_export=false
[0]  ChiTech number of arguments supplied: 4
[0]  Computing cell-centroids.
[0]  Done computing cell-centroids.
[0]  Checking cell-center-to-face orientations
[0]  Done checking cell-center-to-face orientations
[0]  00:00:00 Establishing cell connectivity.
[0]  00:00:00 Vertex cell subscriptions complete.
[0]  00:00:00 Surpassing cell 1000 of 10000 (10%)
[0]  00:00:00 Surpassing cell 2000 of 10000 (20%)
[0]  00:00:00 Surpassing cell 3001 of 10000 (30%)
[0]  00:00:00 Surpassing cell 4000 of 10000 (40%)
[0]  00:00:00 Surpassing cell 5000 of 10000 (50%)
[0]  00:00:00 Surpassing cell 6001 of 10000 (60%)
[0]  00:00:00 Surpassing cell 7001 of 10000 (70%)
[0]  00:00:00 Surpassing cell 8000 of 10000 (80%)
[0]  00:00:00 Surpassing cell 9000 of 10000 (90%)
[0]  00:00:00 Surpassing cell 10000 of 10000 (100%)
[0]  00:00:00 Establishing cell boundary connectivity.
[0]  00:00:00 Done establishing cell connectivity.
[0]  MeshGenerator: Cells created = 10000
[0]  NonLinearSolver non-linear iteration    0 - Residual 9.9503768e+04
[0]  NonLinearSolver iteration    0 - Residual 9.9503768e+04
[0]  NonLinearSolver iteration    1 - Residual 1.0473317e+03
[0]  NonLinearSolver iteration    2 - Residual 2.5399840e+01
[0]  NonLinearSolver iteration    3 - Residual 3.2208154e+00
[0]  NonLinearSolver iteration    4 - Residual 2.7556890e-01
[0]  NonLinearSolver non-linear iteration    1 - Residual 3.3084385e+00
[0]  NonLinearSolver iteration    0 - Residual 3.3084385e+00
[0]  NonLinearSolver iteration    1 - Residual 5.1850104e-02
[0]  NonLinearSolver iteration    2 - Residual 3.9668534e-03
[0]  NonLinearSolver iteration    3 - Residual 3.3829864e-04
[0]  NonLinearSolver iteration    4 - Residual 2.9556466e-05
[0]  NonLinearSolver non-linear iteration    2 - Residual 3.1095642e-05
[0]  
[0]  SCALAR post-processors history at event "ProgramExecuted"
[0]  *--------*-----------*
[0]  | Time   | avgval    |
[0]  *--------*-----------*
[0]  | Latest | 46.884804 |
[0]  *--------*-----------*
[0]  
[0]  SCALAR post-processors latest values at event "ProgramExecuted"
[0]  *----------------------*-----------------*
[0]  | Post-Processor Name  | Value           |
[0]  *----------------------*-----------------*
[0]  | avgval(latest)       |       46.884804 |
[0]  *----------------------*-----------------*
[0]  


